### PR TITLE
Fix  excel-related docstring errors 

### LINKED ISSUE #23494: DOC: Fix docstrings of Excel related functions
The next Excel related functions contain many errors as reported by `scripts/validate_docstrings.py`:
- pandas.read_excel
- pandas.io.formats.style.Styler.to_excel
- Series.to_excel
- DataFrame.to_excel

We should fix them, following our standards: https://pandas.pydata.org/pandas-docs/stable/contributing_docstring.html

The validation script should not report errors after they are fixed.

### PR DESCRIPTION
**Pull request by thoo**
- [x] closes #23494
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] passes/runs `python make.py --single pandas.related-doc`

Fix errors which are  reported by `scripts/validate_docstrings.py`:
 
 * pandas.read_excel
 * pandas.io.formats.style.Styler.to_excel
 * Series.to_excel
 * DataFrame.to_excel

### CODE REVIEW

File: pandas/io/excel.py [hunk 1]
```diff
@@ -166,96 +190,89 @@
...
->>> pd.read_excel('tmp.xlsx')
-      Name  Value
-0  string1      1
-1  string2      2
-2  string3      3
+>>> pd.read_excel('tmp.xlsx', index_col=0)  # doctest: +SKIP
+       Name  Value
+0   string1      1
+1   string2      2
+2  #string3      3
```
Comment (jreback) [inline pandas/io/excel.py]: this hash on purpose?

**Comment by jreback** [comment]
@datapythonista @WillAyd @gfyoung if any comments.

File: pandas/io/excel.py (hunk 1 above)
Comment (thoo) [reply pandas/io/excel.py]: It is for the last example. `pd.read_excel('tmp.xlsx', index_col=0, comment='#')`.

**Review by gfyoung — APPROVED** [review]

**Comment by thoo** [comment]
`python make.py --single pandas.read_excel` stops building docs after this commit 51c508c (merge upstream)

**Comment by jreback** [comment]
@thoo, @datapythonista just changed this a bit, can you post what happens?

**Comment by thoo** [comment]
I got errors like `/Users/johndoe/pandas/doc/source/index.rst:110: WARNING: toctree references excluded document 'generated/pandas.read_excel'` when I run `python make.py --single pandas.read_excel`. 
I think the extensions are not built. When I run `python setup.py build_ext --inplace -j 4`, I just get `running build_ext` and no other messages/output as usual.

**Comment by datapythonista** [comment]
@thoo I just built `pandas_excel` on master (in Linux):
```
$ ./doc/make.py html --single=pandas.read_excel
Running Sphinx v1.8.2
making output directory...
[autosummary] generating autosummary for: index.rst
[autosummary] generating autosummary for: /home/mgarcia/src/pandas/doc/source/generated/pandas.read_excel.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 2 added, 0 changed, 0 removed
reading sources... [100%] index                                                                                                                                                                                                               
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                                                                                                                                                                                
generating indices... genindex py-modindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded.

The HTML pages are in doc/build/html.
```

I think we do a clean before the build, and if I'm not wrong I left the clean of the `doc/source/generated/` directory, but may be you can try to remove it manually if it exists.

**Comment by datapythonista** [comment]
Ups, sorry, my bad. I didn't have the last changes, looks like building a single API doc is broken after splitting `api.rst` that has been just merged. Let me take a look...

**Comment by thoo** [comment]
@datapythonista Thanks. I will remove `doc/source/generated/`.

**Review by datapythonista — COMMENTED** [review]
just one minor thing, other than that lgtm. Thanks @thoo

**Comment by jreback** [comment]
@thoo if you'd pull from master, you should be able to build single pages now.

**Comment by thoo** [comment]
@jreback @datapythonista  Thanks. It is working.

File: pandas/io/excel.py [hunk 2]
```diff
@@ -39,71 +39,69 @@
...
-    Specify None to get all sheets.
+    Available cases:
 
-    str|int -> DataFrame is returned.
-    list|None -> Dict of DataFrames is returned, with keys representing
-    sheets.
+    * Defaults to ``0`` : 1st sheet as a `DataFrame`
+    * ``1`` : 2nd sheet as a `DataFrame`
+    * ``"Sheet1"`` : Load sheet with name "Sheet1"
+    * ``[0,1,"Sheet5"]`` : Load first, second and sheet named "Sheet5"
```
[the remark this answers is not in the document]
Comment (thoo) [reply pandas/io/excel.py]: Done.

File: pandas/io/excel.py (hunk 1 above)
Comment (thoo) [reply pandas/io/excel.py]: Change from `#string3` to `#Comment` to avoid confusion.

**Comment by thoo** [comment]
It looks like  `pytest pandas/tests/io/test_excel.py`  doesn't pass on master. So hopefully that is not related to this PR.

**Comment by datapythonista** [comment]
there are some tests that check the content of the docstrings, so it's probably related to it, I'd take a look

**Pull request merged**